gotrue/_sync/gotrue_client.py

```diff
@@ -399,7 +401,7 @@ def sign_in_with_otp(
                     },
                 },
                 redirect_to=email_redirect_to,
-                xform=parse_auth_response,
+                xform=parse_auth_otp_response,
```

Comment:
Thank you! This PR looks fine to me. 

Do you mind moving the changes to the `_async` folder and then run `make build_sync` after that so that the changes propagate to the sync folder?  This will ensure that the changes here are available on both async and sync methods. I'll approve after that is done

Reply:
Thanks for the remark. Done.